### DISCUSSION TITLE
wiki.js sometimes deletes pages that were changed via git

### DISCUSSION THREAD

**Discussion by twoi** [requarks/wiki#4332] [Error / Bug Report]
Having wiki.js sync with git is a great feature, because it allows one to use the power of the command line to do lots of changes. So I set up a git repo and let wiki.js sync to it. Changes in wiki.js get properly reflected in git. When I change files in git and push, they sometimes get imported properly in wiki.js, but sometimes wiki.js deletes the changed page for reasons I don't know yet. The log will show something like:
```
wiki_1      | 2021-08-21T22:13:51.873Z [MASTER] info: Syncing with storage target git...
wiki_1      | 2021-08-21T22:13:51.885Z [MASTER] info: (STORAGE/GIT) Performing pull rebase from origin on branch main...
wiki_1      | 2021-08-21T22:13:53.157Z [MASTER] info: (STORAGE/GIT) Performing push to origin on branch main...
wiki_1      | 2021-08-21T22:13:53.531Z [MASTER] info: (STORAGE/GIT) Page marked as deleted: Our-Wiki/テンプレート.md
wiki_1      | Loading configuration from /wiki/config.yml... OK
wiki_1      | 2021-08-21T22:13:53.699Z [JOB] info: Rebuilding page tree...
wiki_1      | 2021-08-21T22:13:54.113Z [JOB] info: Using database driver pg for postgres [ OK ]
wiki_1      | 2021-08-21T22:13:54.226Z [JOB] info: Rebuilding page tree: [ COMPLETED ]
```
This happened for all 114 files in a commit that I did. So in wiki.js, all those pages were just gone - quite the disaster for the real company wiki!
Luckily, an "Import Everything" in admin / storage / git got all pages back:
```
wiki_1      | 2021-08-21T22:16:48.004Z [MASTER] info: (STORAGE/GIT) Processing Our-Wiki/テンプレート.md...
wiki_1      | 2021-08-21T22:16:48.007Z [MASTER] info: (STORAGE/GIT) Page marked as new: Our-Wiki/テンプレート.md
wiki_1      | Loading configuration from /wiki/config.yml... OK
wiki_1      | 2021-08-21T22:16:48.226Z [JOB] info: Rendering page ID 426...
wiki_1      | 2021-08-21T22:16:48.593Z [JOB] info: Using database driver pg for postgres [ OK ]
wiki_1      | 2021-08-21T22:16:49.611Z [JOB] info: Rendering page ID 426: [ COMPLETED ]
```
However, now the history of all those 114 pages in wiki.js is gone!

This means I will have to disable git storage, as the history is something I want to keep, and of course having to manually (find and then) restore files deleted by wiki.js is not a workable solution.

**Comment by twoi** [comment]
N.B. I don't think it is related, but in order for files with non-ASCII characters to get printed as Unicode in the wiki logs, rather than octal escapes, I had done this:
```
wiki@wiki:~/wiki$ docker-compose exec wiki bash
bash-5.0$ cd /wiki/data/repo/
bash-5.0$ git config core.quotepath off
```

**Reply by EricFromCanada** [reply]
implemented in https://github.com/requarks/wiki/pull/5272

**Comment by EricFromCanada** [comment]
I believe this issue is fixed with #5272. With quotepath left enabled, non-ASCII characters in the filename would be escaped in git's output, causing the storage engine to not find the file on-disk, leading to this [`!fileExists`](https://github.com/requarks/wiki/blob/73af37b81baac1df75fc343deb82d818cb1eaee8/server/modules/storage/git/storage.js#L202) condition being activated.
